src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortClause.java

```diff
@@ -85,7 +85,26 @@ public boolean hasVectorizeSearchClause() {
             .equals(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD);
   }
 
+  public SortExpression bm25SearchExpression() {
+    if ((sortExpressions == null) || sortExpressions.size() != 1) {
+      return null;
+    }
+    SortExpression expr = sortExpressions.get(0);
+    return expr.isBM25Search() ? expr : null;
+  }
+
   public void validate(CollectionSchemaObject collection) {
+    // First things first: BM25 search may or may not be available
```

Comment:
Availability not checked by deserializer since it has no context of SchemaObject (Collection) -- only available at this point.